### ISSUE TITLE
a conf file per consumer?

### ISSUE THREAD

**Issue by ghost** [ricbra/rabbitmq-cli-consumer#7]
is there a way to have 1 file with credentials instead of in every consumer config file?

**Comment by ricbra** [comment]
Currently there isn't.

I wonder how exactly you'd like to see this feature. A separate file for credentials and add a new option to pass it on to the consumer?

**Comment by appeltaert** [comment]
You dont want credentials as options and you dont want lose the flexibility of a conf per consumer. 

How about the option to pass mutliple conf files so they get merged? Seems to be the most flexible option since you can have multiple hosts and still 1 conf with per host with credentials. This is also backwards compatible wit the current build.

**Comment by ricbra** [comment]
We certainly don't want to loose the flexibility of a config per consumer, so no worries thats not going to happen. Credentials as an option is too insecure (password can be seen by all users on the server using ps or something similar).  

Solution 1:
What about adding a optional <code>--credentials</code> option to pass in a file with the credentials? If given, the credentials in the <code>--credentials</code> file would override any credentials in the <code>--configuration</code> file. If credentials are found in both the <code>--configuration</code> and  <code>--credentials</code> file, a warning should be shown to the user. This could lead to hard to debug cases, so we want to prevent that. This preserves BC. An example of how this could be called:

```
$ rabbitmq-cli-consumer --configuration=config.conf --credentials=credentials.conf 
```

Solution 2:
Take the merging of configuration files even further and provide multiple levels of configuration (just as MySQL does: /etc/my.conf, .my.conf in home dir etc):
- global config in <code>/etc/rabbitmq-cli-consumer/rabbitmq-cli-consumer.conf</code>
- possibility to override in home dir of executing user with <code>~/.rabbitmq-cli-consumer.conf</code>
- pass in config to override with currently available <code>--configuration</code> option.

This also preserves BC.

I think I like solution 2 more then solution 1, what about you @appeltaert ? Too bad the ticket author has deleted his account (or has it something to do with the ddos on Github?)

**Comment by appeltaert** [comment]
Eh ghost? Idk what happened but that comment was mine :) Ye i agree solution 2 is excellent

**Comment by ricbra** [comment]
Funny, the first comment was originally places as "ecoli1". Weird shirt.

I'll add this as feature request.

**Comment by alex-pex** [comment]
Hello,

The only option which annoys me is the "queue" setting. I would like not to write the value in the config file and overwrite it at execution.

Something like : 

``` shell
$ rabbitmq-cli-consumer --configuration=config.conf --executable="php app/console test:event" --queue=hello
```

That way I can reuse the same config for several consumers

**Comment by ricbra** [comment]
I've been thinking about this problem past few days and I think it would be best if we (as a final resort) also could pass in all configuration options (where possible) as command line arguments. That provides ultimate flexibility. 

tl;dr
- global config in <code>/etc/rabbitmq-cli-consumer/rabbitmq-cli-consumer.conf</code>
- possibility to override in home dir of executing user with <code>~/.rabbitmq-cli-consumer.conf</code>
- pass in config to override with currently available <code>--configuration</code> option.
- pass in argument per option to override specific argument(s)

**Comment by ricbra** [comment]
I'm closing this as it is implemented in 2.0 branch.
